rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -567,8 +590,23 @@ rmw_create_publisher(
       allocator->deallocate(const_cast<char *>(rmw_publisher->topic_name), allocator->state);
     });
 
+  // Convert the type hash to a string so that it can be included in
+  // the keyexpr.
+  char * type_hash_c_str = nullptr;
+  rcutils_ret_t stringify_ret = rosidl_stringify_type_hash(
+    publisher_data->type_hash,
+    *allocator,
+    &type_hash_c_str);
+  if (RCUTILS_RET_BAD_ALLOC == stringify_ret) {
+    RMW_SET_ERROR_MSG("Failed to allocate type_hash_c_str.");
+    return nullptr;
+  }
+
```

Comment:
We need to free the `type_hash_c_str` when we are done with it, something like:

```
  auto free_type_hash_c_str = rcpputils::make_scope_exit(
    [&type_hash_c_str]() {
      allocator.deallocate(type_hash_c_str, allocator.state);
    });
```

Reply:
Done https://github.com/ros2/rmw_zenoh/pull/171/commits/5796e7426454f4ac73ce37a28a00954cf327c32c